India EDM: State-wise Energy Generation and Consumption data

SenthamizhanV opened the pull request:
PR containing energy consumption and generation data for Indian states

SenthamizhanV commented:
@ajaits I have mentioned new Enums and ensured the dc lints pass successfully. I have also mentioned the new Enums in the [import doc](https://docs.google.com/document/d/1Dw_i_mkS2bT4Oc80s52oz3zEKvKDIIuFrulqlWxHJm0/edit?usp=sharing&resourcekey=0-AfIWrbZOZOYgIttUDuXL1Q). Please take a look at the PR.

SenthamizhanV commented on scripts/india_edm/IndiaEnergy_Coal/test_data/test.mcf:
```diff
@@ -0,0 +1,180 @@
+measuredProperty: dcs:consumption
+measurementQualifier: dcs:Annual
+statType: dcs:measuredValue
+energySource: dcs:CoalProducts
```
Yes, these properties work for this SV. But there are other SVs like [Annual_Consumption_Coal_ConstructionIndustry_CoalProducts](https://github.com/datacommonsorg/data/blob/ea13e41d8fbac17e454b561d8bc8e4c726ca4368/scripts/india_edm/IndiaEnergy_Coal/test_data/test.mcf#L37) that find its application in Construction Industry (cement production in particular). Should we change the populationType for each SV then? In that case, populationType should be related to the consumingSector - like it will be `populationType: Product` for this SV.

If I may suggest, can we do `populationType: Energy` for all fuel sources in this dataset? Because all types of coal consumption result in generating Energy. I think that will be a more appropriate value.

beets replied:
Yes, that makes sense. Where we can use Electricity, or a more specific type, we should use that (like in this example). For consumption, keeping it as `populationType: Coal` seems appropriate.

ajaits commented on scripts/india_edm/IndiaEnergy_Gas/IndiaEnergy_Gas.mcf:
```diff
@@ -0,0 +1,167 @@
+measuredProperty: dcs:consumption
+measurementQualifier: dcs:Annual
+statType: dcs:measuredValue
+energySource: dcs:NaturalGas
```
Can you pls use the property `fuelType: dcs:NaturalGas` since populationType is Fuel.
This will also match existing statvars like:
[Annual_Consumption_Fuel_Households_NaturalGas`](https://autopush.datacommons.org/browser/Annual_Consumption_Fuel_Households_NaturalGas)

SenthamizhanV replied:
Added the `fuelType` property for Gas [mcf](https://github.com/datacommonsorg/data/blob/5210c3d67ac59d7ed771fff3330e8068f3e6f74e/scripts/india_edm/IndiaEnergy_Gas/IndiaEnergy_Gas.mcf#L7).

Also, resolved your other comments - removed `dc_generated/` folders and zipped all data into one file. Also added a functionality in the code that will automatically unzip the data files and use it in the preprocessing.

ajaits reviewed (commented):
Can you pls remove all the `dc_generated/` folders from the PR? They will be created when menifesting the data internally.

ajaits reviewed (commented):
Can you pls zip all the source files under data/ into a single archive and save it as a GitLFS file?

beets reviewed (commented):
Thanks for the clear configs and documentation! The code is also very nicely structured and easy to follow.

Please add the data directory to git lfs (https://git-lfs.github.com/ for instructions)

And let me know if we should revisit the discussion on the SV population types.

Thanks!

SenthamizhanV commented on scripts/india_edm/README.md:
```diff
@@ -0,0 +1,40 @@
+
+## Script usage
+
+Run the `preprocess` module inside the `IndiaEnergy_*` directories to create the MCF, TMCF and preprocessed CSV files of the corresponding dataset.
```
Removed the CSV, MCF and TMCF files from the repo.

SenthamizhanV commented on scripts/india_edm/base.py:
```diff
@@ -0,0 +1,366 @@
+        corresponding dcids.
+        """
+
+        util_path = os.path.join(module_dir, 'util/')
```
Removed the trailing slash.

SenthamizhanV commented on scripts/india_edm/base.py:
```diff
@@ -0,0 +1,366 @@
+            self.js_statvars = json.loads(statvars.read())
+
+        with open(util_path + self.json_file, 'r') as types:
+            self.js_types = json.loads(types.read())
```
Created new variables for each path using `os.path.join` and used those in the handlers.

SenthamizhanV commented on scripts/india_edm/base.py:
```diff
@@ -0,0 +1,366 @@
+        with zipfile.ZipFile(zipfile_path, 'r') as zip_ref:
+            zip_ref.extractall(extracted_path)
+
+        data_path = os.path.join(module_dir, 'data/{}/'.format(self.cat))
```
Implemented both changes

SenthamizhanV commented:
> Thanks for the clear configs and documentation! The code is also very nicely structured and easy to follow.
> 
> Please add the data directory to git lfs (https://git-lfs.github.com/ for instructions)
> 
> And let me know if we should revisit the discussion on the SV population types.
> 
> Thanks!

@beets I changed just one populationType in the import. For Generation in Coal, I have changed the popType from Coal to Electricity since it is the product that is being generated from Coal ([commit diff](https://github.com/datacommonsorg/data/pull/619/commits/6fba4bf911aba03fa8367ddc617b53aca266039a)).

Apart from that, all other props and popTypes look reasonable and relevant.

beets reviewed (approved):
thanks - this looks good!
